SConstruct

```diff
@@ -656,40 +656,37 @@ elif methods.using_gcc(env):
             "to switch to posix threads."
         )
         Exit(255)
-    if env["debug_paths_relative"] and cc_version_major < 8:
-        print_warning("GCC < 8 doesn't support -ffile-prefix-map, disabling `debug_paths_relative` option.")
-        env["debug_paths_relative"] = False
 elif methods.using_clang(env):
     # Apple LLVM versions differ from upstream LLVM version \o/, compare
     # in https://en.wikipedia.org/wiki/Xcode#Toolchain_versions
+    vanilla = True
+
     if env["platform"] == "macos" or env["platform"] == "ios":
         vanilla = methods.is_vanilla_clang(env)
```

Comment:
I also though about turning it into one line of code `vanilla = methods.is_vanilla_clang(env)` because I assume it will return `False` if we are not at mac but I doubt...

Reply:
It will return `False` if you are not building for macOS/iOS, the function is simply checking if clang version string have `Apple` in it.

Reply:
I know but it feels somewhat strange and hacky. If method was called `is_apple_clang` which is more good name it would contain semantic of `if platform == "macos"` in the name. Now if you  fold it to one line it looses its semantic... And I doubt that renaming it is allowed because someone who builds custom godot for macos with a big chance relies on it. If I am allowed to rename it I will gladly do it

Reply:
I think it's ok to rename it, compatibility is important for documented public API, but this is an internal build system method.